crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -88,10 +88,9 @@ impl Server {
     pub async fn run(mut self) -> Result<(), Error> {
         get_or_init_server_start_time().await;
 
-        // Compatibility check
-        info!("Starting compatibility check");
-        check_all_tables(&self.db_reader).await?;
-        info!("Compatibility check passed");
+        // Make sure to run all migrations on startup, and also serve as a compatibility check.
+        run_migrations(&mut get_pool_connection(&self.db_reader.inner.get_pool())?)
```

Comment:
I don't think GraphQL (or any reader) should run migrations directly -- just check that they have run. GraphQL's connection to the DB should not generally have write permissions, and I think that's a good property to preserve to avoid mistakes.

Reply:
Agree with this. We should even think about giving our readers (graphql, jsonrpc) read-only user roles on the DB so that they cannont write to the db